src/tools/miri/tests/pass/static_align.rs

```diff
@@ -0,0 +1,50 @@
+// Test that miri respects the `target.min_global_align` value for the target.
+// E.g. on s390x, statics have, in practice, a minimum alignment of 2.
+
+static T0: bool = true;
+static T1: bool = true;
+static T2: bool = true;
+static T3: bool = true;
+static T4: bool = true;
+static T5: bool = true;
+static T6: bool = true;
+static T7: bool = true;
+static T8: bool = true;
+static T9: bool = true;
+
+static F0: bool = false;
+static F1: bool = false;
+static F2: bool = false;
+static F3: bool = false;
+static F4: bool = false;
+static F5: bool = false;
+static F6: bool = false;
+static F7: bool = false;
+static F8: bool = false;
+static F9: bool = false;
+
+fn main() {
+    let min_align = if cfg!(target_arch = "s390x") { 2 } else { 1 };
+
+    assert!(((&T0) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T1) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T2) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T3) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T4) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T5) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T6) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T7) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T8) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&T9) as *const _ as usize).is_multiple_of(min_align));
+
+    assert!(((&F0) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F1) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F2) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F3) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F4) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F5) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F6) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F7) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F8) as *const _ as usize).is_multiple_of(min_align));
+    assert!(((&F9) as *const _ as usize).is_multiple_of(min_align));
```

Comment:
It would be nice to have `.addr()` here instead of ptr-as-usize. No need to hurt yourself doing the change though, I think we have plenty of Miri tests already that use `as usize` when they could do something else.

Reply:
`ptr::from_ref(&ABC123).addr()`